File: ai/stream/sse_a01.js
const EventSource = require('eventsource');


const url = 'https://api.example.com/llm/sse-stream';

const eventSource = new EventSource(url, {
  headers: {
    'Authorization': 'Bearer YOUR_API_KEY',
    'Content-Type': 'application/json',
  },
  method: 'POST',
  body: JSON.stringify({ prompt: "Tell me a long story" }),
});

eventSource.onmessage = (event) => {
  console.log(event.data);
};

eventSource.onerror = (error) => {
  console.error('Error:', error);
  eventSource.close();
};

File: ai/stream/sse_a01.py
#!/usr/bin/env python3

import requests


def stream_llm_sse():
    url = "https://api.example.com/llm/sse-stream"
    headers = {
        "Accept": "text/event-stream",
        "Authorization": "Bearer YOUR_API_KEY",
    }
    data = {"prompt": "Tell me a long story"}
    
    response = requests.post(url, headers=headers, json=data, stream=True)
    
    for line in response.iter_lines():
        if line:
            decoded_line = line.decode('utf-8')
            if decoded_line.startswith("data:"):
                print(decoded_line[5:].strip())

